pkg/blob/nodeserver_test.go

```diff
@@ -935,13 +937,72 @@ func TestNodeExpandVolume(t *testing.T) {
 	}
 }
 
+// fakeMountServiceServer implements mount_azure_blob.MountServiceServer for testing
+type fakeMountServiceServer struct {
+	mount_azure_blob.MountServiceServer
+	mockOutput string
+	mockError  error
+}
+
+// MountAzureBlob implements the mount service method
+func (s *fakeMountServiceServer) MountAzureBlob(ctx context.Context, req *mount_azure_blob.MountAzureBlobRequest) (*mount_azure_blob.MountAzureBlobResponse, error) {
+	return &mount_azure_blob.MountAzureBlobResponse{
+		Output: s.mockOutput,
+	}, s.mockError
+}
+
 func TestMountBlobfuseWithProxy(t *testing.T) {
+	// Create a mock gRPC server and client for better test coverage
+
+	// Create a test server with our implementation
+	mockServer := &fakeMountServiceServer{
+		mockOutput: "mock mount successful",
+		mockError:  nil,
+	}
+
+	// Start a local gRPC server for testing
+	lis, err := net.Listen("tcp", "127.0.0.1:0") // Use random available port
+	if err != nil {
+		t.Fatalf("Failed to listen: %v", err)
+	}
+	s := grpc.NewServer()
+	mount_azure_blob.RegisterMountServiceServer(s, mockServer)
+
+	// Start the server in a goroutine
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Logf("Server exited with error: %v", err)
+		}
+	}()
+	defer s.Stop() // Ensure server is stopped after test
+
+	// Run test case 1: Success case with mock server
 	args := "--tmp-path /tmp"
 	authEnv := []string{"username=blob", "authkey=blob"}
 	d := NewFakeDriver()
-	_, err := d.mountBlobfuseWithProxy(args, "fuse", authEnv)
-	// should be context.deadlineExceededError{} error
-	assert.NotNil(t, err)
+	
+	// Point to our mock server
+	d.blobfuseProxyEndpoint = lis.Addr().String()
+	d.blobfuseProxyConnTimout = 5 // 5 second timeout
+
+	// Test the success path
+	output, err := d.mountBlobfuseWithProxy(args, "fuse", authEnv)
+	assert.NoError(t, err)
+	assert.Equal(t, "mock mount successful", output)
+
+	// Test case 2: Error from MountAzureBlob
+	mockServer.mockError = fmt.Errorf("mock mount error")
+	output, err = d.mountBlobfuseWithProxy(args, "fuse", authEnv)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "mock mount error")
+	
+	// Test case 3: Connection error case (using non-existent endpoint)
```

Comment:
is it possible to use test case array like this?
https://github.com/kubernetes-sigs/blob-csi-driver/blob/42b05af847c3adb7c8c1742807c3ba8513439362/pkg/blob/azure_test.go#L429-L454